server/src/thread/model.rs

```diff
@@ -350,6 +351,95 @@ impl Threads {
 
         Ok(comment)
     }
+
+    pub async fn update_comment<'a, E>(
+        payload: &update_comment_mutation::Payload,
+        executor: E,
+    ) -> Result<operation::SuccessOutput, operation::Error>
+    where
+        E: Executor<'a>,
+    {
+        if payload.content.is_empty() {
+            return Err(operation::Error::BadRequest {
+                reason: "content is empty".to_string(),
+            });
+        }
+
+        let mut transaction = executor.begin().await?;
+
+        let is_trashed = match sqlx::query!(
+            r#"
+                SELECT trashed FROM uuid WHERE id = ?
```

Comment:
You can fetch this info in only one query, joining this query with the one in line 395.
use a join
``SELECT ...,, trashed FROM comment JOIN uuid ON uuid.id = comment.id ... `